### PR TITLE
Refactored hook handlers in mod_disco

### PR DESCRIPTION
**Pull request by pawlooss1**
This PR changes all hook handlers in `mod_disco` module to `gen_hook` format.

### CODE REVIEW

**Comment by mongoose-im** [comment]
[small_tests_24](https://circleci.com/gh/esl/MongooseIM/150080) / small_tests / b4f7bbdd4db857f67661b3ad3a105e159e90f165
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/3847/150080/small_tests.24.3.4) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/3847/150080/small_tests.24.3.4/small/ct_run.mongooseim@localhost.2022-11-08_10.39.57/lib.mongooseim.logs/run.2022-11-08_10.39.57/suite.log.html)

File: src/mod_disco.erl [hunk 1]
```diff
@@ -197,57 +193,74 @@ process_sm_iq_info(Acc, From, To, #iq{type = get, lang = Lang, sub_el = SubEl} =
...
     {ok, Acc}.
 
 %% @doc Support for: XEP-0157 Contact Addresses for XMPP Services
--spec disco_info(mongoose_disco:info_acc()) -> mongoose_disco:info_acc().
-disco_info(Acc = #{host_type := HostType, module := Module, node := <<>>}) ->
+-spec disco_info(Acc, Params, Extra) -> {ok, Acc} when
+    Acc :: mongoose_disco:info_acc(),
+    Params :: map(),
+    Extra :: map().
+disco_info(Acc = #{host_type := HostType, module := Module, node := <<>>}, _, _) ->
```
Comment (NelsonVides) [inline src/mod_disco.erl:257]: Hmmm... all these handlers expose the host_type twice, once here and once in the extra param, seems like redundancy to correct for the future 🤔

File: src/mod_disco.erl [hunk 2]
```diff
@@ -197,57 +193,74 @@ process_sm_iq_info(Acc, From, To, #iq{type = get, lang = Lang, sub_el = SubEl} =
...
+    Acc :: mongoose_disco:identity_acc(),
+    Params :: map(),
+    Extra :: map().
+disco_local_identity(Acc = #{node := <<>>}, _, _) ->
+    {ok, mongoose_disco:add_identities([#{category => <<"server">>,
                                      type => <<"im">>,
-                                     name => <<"MongooseIM">>}], Acc);
-disco_local_identity(Acc) ->
-    Acc.
+                                     name => <<"MongooseIM">>}], Acc)};
```
Comment (NelsonVides) [inline src/mod_disco.erl:203]: Indentation broken.

**Review by NelsonVides — COMMENTED** [review]
Just one indentation broken and it's ready to merge 👌🏽

File: src/mod_disco.erl (hunk 2 above)
Comment (pawlooss1) [reply src/mod_disco.erl]: done 👍

**Comment by mongoose-im** [comment]
[small_tests_24](https://circleci.com/gh/esl/MongooseIM/150348) / small_tests / caee162510545023b7efe56009aa1f0c1e285c7f
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/3847/150348/small_tests.24.3.4) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/3847/150348/small_tests.24.3.4/small/ct_run.mongooseim@localhost.2022-11-09_08.44.49/lib.mongooseim.logs/run.2022-11-09_08.44.49/suite.log.html)

**Review by NelsonVides — APPROVED** [review]

**Pull request merged**